Library/Formula/cfssl.rb

```diff
@@ -0,0 +1,103 @@
+  end
+  go_resource "github.com/cloudflare/cf-tls" do
+    url "https://github.com/cloudflare/cf-tls.git",
+      :revision => "358b61f346ff4384bd2b051a8d1a24c48444e388"
+  end
+  go_resource "github.com/daaku/go.zipexe" do
+    url "https://github.com/daaku/go.zipexe.git",
+      :revision => "a5fe2436ffcb3236e175e5149162b41cd28bd27d"
+  end
+  go_resource "github.com/dgryski/go-rc2" do
+    url "https://github.com/dgryski/go-rc2.git",
+      :revision => "8a9021637152186df738b1ec376caf2100fef194"
+  end
+  go_resource "github.com/miekg/pkcs11" do
+    url "https://github.com/miekg/pkcs11.git",
+      :revision => "4b4363b37cd397baa396bcde75f4b09f40441bd8"
+  end
+  go_resource "github.com/kardianos/osext" do
+    url "https://github.com/kardianos/osext.git",
+      :revision => "29ae4ffbc9a6fe9fb2bc5029050ce6996ea1d3bc"
+  end
+  go_resource "github.com/coreos/go-systemd" do
+    url "github.com/coreos/go-systemd.git",
+    :revision => "2ed5b5012ccde5f057c197890a2c801295941149"
+  end
+
+  depends_on "go" => :build
+  depends_on "libtool" => :run if build.with? "pkcs11"
+
+  def install
+    ENV["GOPATH"] = buildpath
+    cfsslpath = buildpath/"src/github.com/cloudflare/cfssl"
+    cfsslpath.install Dir["{*,.git}"]
+    Language::Go.stage_deps resources, buildpath/"src"
+    cd "src/github.com/cloudflare/cfssl" do
+      if build.with? "pkcs11"
+        system "go", "build", "-o", "build/cfssl", "cmd/cfssl/cfssl.go"
+      else
+        system "go", "build", "-tags", "nopkcs11", "-o", "build/cfssl", "cmd/cfssl/cfssl.go"
+      end
+      system "go", "build", "-o", "build/cfssljson", "cmd/cfssljson/cfssljson.go"
+      system "go", "build", "-o", "build/mkbundle", "cmd/mkbundle/mkbundle.go"
+      bin.install "build/cfssl", "build/cfssljson", "build/mkbundle"
+    end
+  end
+
+  test do
+    begin
+      require "utils/json"
+      File.open(testpath/"request.json", "w") do |file|
+        file.write(
+          Utils::JSON.dump(
+            :CN => "Your Certificate Authority",
+            :hosts => [],
+            :key => {
+              :algo => "rsa",
+              :size => 4096,
+            },
+            :names => [
+              {
+                :C => "US",
+                :ST => "Your State",
+                :L => "Your City",
+                :O => "Your Organization",
+                :OU => "Your Certificate Authority",
+              },
+            ]
+          )
+        )
+      end
+      shell_output("#{bin}/cfssl genkey -initca request.json > response.json")
+      response = Utils::JSON.load(File.read(testpath/"response.json"))
+      assert_match /^-----BEGIN CERTIFICATE-----.*/, response["cert"]
+      assert_match /.*-----END CERTIFICATE-----$/, response["cert"]
+      assert_match /^-----BEGIN RSA PRIVATE KEY-----.*/, response["key"]
+      assert_match /.*-----END RSA PRIVATE KEY-----$/, response["key"]
+      File.delete(testpath/"request.json")
+      File.delete(testpath/"response.json")
```

Comment:
You don’t need to delete the files; the test is executed in a temporary directory.